src/infra/core/src/testing/dataset_data_helper.rs

```diff
@@ -29,6 +32,20 @@ impl DatasetDataHelper {
         Self { dataset, ctx }
     }
 
+    pub async fn data_slice_count(&self) -> usize {
+        let v: Vec<_> = self
+            .dataset
+            .as_metadata_chain()
+            .iter_blocks()
+            .filter_data_stream_blocks()
+            .filter_map_ok(|(_, b)| b.event.new_data.map(|_| 1))
+            .try_collect()
+            .await
+            .unwrap();
+
+        v.iter().sum()
```

Comment:
`tokio_stream::stream_ext::StreamExt` have `fold()` and maybe more importantly `try_fold()`.

I think this might be a better solution, especially for IoT cases where we can have a potentially huge number of blocks (hence a huge allocated vector)

Reply:
you got me, I got annoyed with trying to make `try_fold` work and just because this is a test code wrote a very lazy solution - revisited now